### PR TITLE
Internal async pool conversion

### PR DESCRIPTION
**Pull request by Kinju011**
Internal=>Async=>Pool directory files converted

### CODE REVIEW

File: ruby/neo4j/driver/internal/async/pool/pool_settings.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            idle_time_before_connection_test
+          end
+
+          def idle_time_before_connection_test_enabled?
+            idle_time_before_connection_test >= 0
+          end
+
+          def max_connection_lifetime
+            max_connection_lifetime
+          end
```
Comment (klobuczek) [inline ruby/neo4j/driver/internal/async/pool/pool_settings.rb:31]: those methods are already defined by `attr_reader`

**Review by klobuczek — CHANGES_REQUESTED** [review]

**Review by klobuczek — CHANGES_REQUESTED** [review]

File: ruby/neo4j/driver/internal/async/pool/connection_pool_impl.rb [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+                  assert_not_closed(address, channel, pool)
+                  Connection::ChannelAttributes.set_authorization_state_listener(channel, channel_health_checker)
+                  connection = connection_factory.create_connection(channel, pool)
+
+                  metrics_listener.after_acquired_or_created(pool.id, acquire_event)
+                  connection
+                ensure
+                  metrics_listener.after_acquiring_or_creating(pool.id)
+                end
+              end
```
Comment (mrhardikjoshi) [inline ruby/neo4j/driver/internal/async/pool/connection_pool_impl.rb:44]: Spacing issue
```suggestion
            channel_future.handle do |channel, error|
              begin
                process_acquisition_error(pool, address, error)
                assert_not_closed(address, channel, pool)
                Connection::ChannelAttributes.set_authorization_state_listener(channel, channel_health_checker)
                connection = connection_factory.create_connection(channel, pool)

                metrics_listener.after_acquired_or_created(pool.id, acquire_event)
                connection
              ensure
                metrics_listener.after_acquiring_or_creating(pool.id)
              end
            end
```

File: ruby/neo4j/driver/internal/async/pool/connection_pool_impl.rb [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+                      entry_iterator.remove
+
+                      unless pool.nil?
+                        log.info("Closing connection pool towards #{address}, it has no active connections and is not in the routing table registry.")
+                        close_pool_in_background(address, pool)
+                      end
+                    end
+                  end
+                end
+              end
```
Comment (mrhardikjoshi) [inline ruby/neo4j/driver/internal/async/pool/connection_pool_impl.rb:67]: This is not aligned correctly

File: ruby/neo4j/driver/internal/async/pool/connection_pool_impl.rb [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+            end
+            close_future
+          end
+
+          def is_open?(address)
+            Util::LockUtil.execute_with_lock(address_to_pool_lock.read_lock, -> () {address_to_pool.contains_key(address)}.call)
+          end
+
+          def to_string
+            Util::LockUtil.execute_with_lock(address_to_pool_lock.read_lock, -> () {"ConnectionPoolImpl{ pools=#{address_to_pool}}"}.call)
```
Comment (mrhardikjoshi) [inline ruby/neo4j/driver/internal/async/pool/connection_pool_impl.rb:103]:
```suggestion
            Util::LockUtil.execute_with_lock(address_to_pool_lock.read_lock) { "ConnectionPoolImpl{ pools=#{address_to_pool}}" })
```

File: ruby/neo4j/driver/internal/async/pool/netty_channel_pool.rb [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+          end
+
+          def is_closed?
+            closed.get
+          end
+
+          def id
+            id
+          end
+
```
Comment (mrhardikjoshi) [inline ruby/neo4j/driver/internal/async/pool/netty_channel_pool.rb:46]: not needed as `id` is marked with `attr_reader` above
```suggestion
```

File: ruby/neo4j/driver/internal/async/pool/netty_channel_pool.rb [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    module Async
+      module Pool
+        class NettyChannelPool
+          # Unlimited amount of parties are allowed to request channels from the pool.
+          MAX_PENDING_ACQUIRES = java.lang.Integer::MAX_VALUE
+
+          # Do not check channels when they are returned to the pool.
+          RELEASE_HEALTH_CHECK = false
+
+          attr_reader :delegate, :closed, :id, :close_future
```
Comment (mrhardikjoshi) [inline ruby/neo4j/driver/internal/async/pool/netty_channel_pool.rb:12]: only `id` was exposed with reader, in original class unless I am missing something here.

```suggestion
          attr_reader :id
```

File: ruby/neo4j/driver/internal/async/pool/pool_settings.rb [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+  module Internal
+    module Async
+      module Pool
+        class PoolSettings
+          attr_reader :max_connection_pool_size, :connection_acquisition_timeout, :max_connection_lifetime, :idle_time_before_connection_test
+
+          NOT_CONFIGURED = -1
+          DEFAULT_MAX_CONNECTION_POOL_SIZE = 100
+          DEFAULT_IDLE_TIME_BEFORE_CONNECTION_TEST = NOT_CONFIGURED
+          DEFAULT_MAX_CONNECTION_LIFETIME = java.util.concurrent.TimeUnit::HOURS.to_millis(1)
```
Comment (mrhardikjoshi) [inline ruby/neo4j/driver/internal/async/pool/pool_settings.rb:11]:
```suggestion
          DEFAULT_MAX_CONNECTION_LIFETIME = 60*60*1000 # in milliseconds
```

File: ruby/neo4j/driver/internal/async/pool/pool_settings.rb [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+          DEFAULT_CONNECTION_ACQUISITION_TIMEOUT = java.util.concurrent.TimeUnit::SECONDS.to_millis(60)
```
Comment (mrhardikjoshi) [inline ruby/neo4j/driver/internal/async/pool/pool_settings.rb:12]:
```suggestion
          DEFAULT_CONNECTION_ACQUISITION_TIMEOUT = 60 * 1000 # in milliseconds
```

File: ruby/neo4j/driver/internal/async/pool/connection_pool_impl.rb [hunk 9]
```diff
@@ -0,0 +1,213 @@
...
+            end
+            close_future
+          end
+
+          def is_open?(address)
+            Util::LockUtil.execute_with_lock(address_to_pool_lock.read_lock, -> () {address_to_pool.contains_key(address)}.call)
+          end
+
+          def to_string
+            Util::LockUtil.execute_with_lock(address_to_pool_lock.read_lock) { "ConnectionPoolImpl{ pools=#{address_to_pool}}" })
```
Comment (mrhardikjoshi) [inline ruby/neo4j/driver/internal/async/pool/connection_pool_impl.rb:103]:
```suggestion
            Util::LockUtil.execute_with_lock(address_to_pool_lock.read_lock) { "ConnectionPoolImpl{ pools=#{address_to_pool}}" }
```

**Pull request merged**